test/core.js

```diff
@@ -60,12 +60,48 @@ QUnit.test( "jQuery( '#' )", function( assert ) {
 	assert.expect( 2 );
 
 	expectWarning( assert, "Selector, through the jQuery constructor, nothing but hash",
-			function() {
-		var set = jQuery( "#" );
-		assert.equal( set.length, 0, "empty set" );
+		function() {
+			var set = jQuery( "#" );
+			assert.equal( set.length, 0, "empty set" );
 	} );
 } );
 
+QUnit.test( "Improperly closed elements", function( assert ) {
+	assert.expect( 4 );
+
+	var oldWindowAlert = window.alert;
+	window.alert = function() {
+		assert.notOk( "Called alert with " + arguments );
+	};
+
+	expectWarning( assert, "Elements not self-closable nested wrong", 4, function() {
+		jQuery( "<div><p/><span/></div>" );
+		jQuery( "<blockquote><p class='bad' /><span/></blockquote>" );
+		jQuery( "<div />" ).append( "<div data-borked='y'><span/> <p/></div>" );
+		jQuery( "<script /><div class=afterscript/>" );
+	} );
+
+	expectNoWarning( assert, "Script doesn't run in processed strings", function() {
+		jQuery( "<script>alert( 'YOU SHOULD NOT SEE THIS ALERT' )</script>" );
+		jQuery( "<div>" ).append( "<script src='YOU-SHOULD-NOT-SEE-THIS.SCRIPT'></script>" );
+		jQuery( "<script>alert( 'YOU SHOULD NOT SEE THIS CONSOLE' )</script>" );
+	} );
```

Comment:
This doesn't hurt but out of curiosity: is this related to the tag self-closing behavior?

Reply:
Mainly I just wanted to make sure we weren't introducing any script issues with this.